addons/account_anglo_saxon/invoice.py

```diff
@@ -170,19 +171,20 @@ def _anglo_saxon_purchase_move_lines(self, cr, uid, i_line, res, context=None):
                                 i_line.price_unit * (1-(i_line.discount or 0.0)/100.0), line['quantity'])['total']
                             price_line = round(valuation_price_unit * line['quantity'], account_prec)
                             price_diff = round(price_unit - price_line, account_prec)
-                            line.update({'price': price_line})
-                            diff_res.append({
-                                'type': 'src',
-                                'name': i_line.name[:64],
-                                'price_unit': round(price_diff / line['quantity'], account_prec),
-                                'quantity': line['quantity'],
-                                'price': price_diff,
-                                'account_id': acc,
-                                'product_id': line['product_id'],
-                                'uos_id': line['uos_id'],
-                                'account_analytic_id': line['account_analytic_id'],
-                                'taxes': line.get('taxes', []),
-                                })
+                            if price_diff != 0.0:
```

Comment:
I added this if statement because otherwise we will get a journal entry with the price difference account with a value of 0.0.